lib/lz4.c

```diff
@@ -374,7 +374,7 @@ static const int      dec64table[8] = {0, 0, 0, -1, -4,  1, 2, 3};
 
 
 #ifndef LZ4_FAST_DEC_LOOP
-#  if defined __i386__ || defined _M_IX86 || defined __x86_64__ || defined _M_X64
+#  if defined _MSC_VER || defined __i386__ || defined _M_IX86 || defined __x86_64__ || defined _M_X64
```

Comment:
2 things :

`_MSC_VER` checks if Visual Studio is being used.

But Visual Studio can be used to compile multiple targets, such as ARM, and the request was limited to enabling it for normal `x86`/`x64` targets only.

Extending beyond that scope is a not forbidden, but it's a different topic, and will have to be backed by a dedicated set of benchmarks.

Second, I note that the `defined` line already tests Visual's own macros `_M_X64` and `_M_IX86`.
It means that the task has already been fulfilled : the fast decoder is already unlocked for Visual Studio.